src/result_processor.c

```diff
@@ -53,9 +53,15 @@ void SearchResult_Destroy(SearchResult *r) {
   SearchResult_Clear(r);
   RLookupRow_Cleanup(&r->rowdata);
 }
-
-static int RPGeneric_NextEOF(ResultProcessor *rp, SearchResult *res) {
-  return RS_RESULT_EOF;
+static int EndIteratorResult_to_RPResult(int rc) {
```

Comment:
```suggestion
static int IteratorResult_to_RPResult(int rc) {
```
Or just free the spec in both cases